### ISSUE TITLE
Firestore - enablePersistence() throws "Firestore has already been started and its settings can no longer be changed." when used with emulators

### ISSUE THREAD

**Issue by GuasaPlay** [nuxt-community/firebase-module#451]
Hello everyone.

![image](https://user-images.githubusercontent.com/45466316/105894939-dbcfac00-5fe2-11eb-97a7-4c059da84f00.png)

> Firestore has already been started and its settings can no longer be changed. You can only modify settings before calling any other methods on a Firestore object.

I have a problem when I try to use the emulator in firestore. In previous issues #390 #116  they have the same problem, I followed the same steps to correct the problem but it still shows me the error.
Maybe you can help me with something that I am doing wrong
This is my config in nuxt.config.js

![image](https://user-images.githubusercontent.com/45466316/105888448-fa31a980-5fda-11eb-8b90-f312d2012461.png)
![image](https://user-images.githubusercontent.com/45466316/105888487-04ec3e80-5fdb-11eb-8779-2d291de90794.png)

**Comment by miketierce** [comment]
I'm still having this issue as well. 

I can technically get it working by defining the settings as such... but if I refresh the browser the error presents itself. Then I have to ctrl + c to kill the app and restart it and it is working again

And I have to remove the setting's logic completely to get it to work in production, but I feel like there should be a way to find the correct host in the ternary operator

```js
// nuxt-config.js
    render: {
      bundleRenderer: {
        runInNewContext: false
      }
    },
    services: {
      auth: {
        ssr: true,
        persistence: 'local',
        initialize: {
          onAuthStateChangedMutation: null,
          onAuthStateChangedAction: 'user/SET_AUTH_USER'
        }
      },
      // ...
      // firestore: true
      firestore: {
        // enablePersistence: true,
        // emulatorPort: process.env.NODE_ENV !== 'production' ? 8080 : undefined,
        // emulatorHost: 'localhost'
        settings: {
          host:
            process.env.NODE_ENV !== 'production'
              ? 'localhost:8080'
              : 'NOT_SURE_HOW_TO_GET_PRODUCTION_HOST?',
          ssl: process.env.NODE_ENV === 'production'
        }
      }
    }
```

**Comment by lupas** [comment]
@GuasaPlay @miketierce 
I can't really reproduce this issue, could you create a sample project on https://template.nuxtjs.org/ (Codesandbox) where this issue appears so I can debug accordingly?

Are you sure you have the newest version of Firebase & nuxtjs/firebase installed and don't import Firebase anywhere else in your project?

**Comment by miketierce** [comment]
Oh my. I think I am importing Firebase into the login component as well. 

I'll try cleaning that up first.

**Comment by GuasaPlay** [comment]
@miketierce could you explain more about this?

**Comment by tibs245** [comment]
I have the same problem as @miketierce. 
When I have "firestore / enablePersistence" option to true (and emulatorHost / emulatorPort configured). 
I have the error message : *Firestore has already been started and its settings can no longer be changed.*

For bypass the problem I set firestore => enablePersistence to false in development mode.

I have try to reproduce it on codesandbox as @lupas asked. But I have a others problem when I install firebase dependance on codesandbox : https://codesandbox.io/s/pedantic-shockley-wluw9?file=/package.json **(Not relevance)**

**Comment by buwilliams** [comment]
+1 This is still an issue.

**Comment by miketierce** [comment]
It turns out I had a few other places in my code that I was still importing and using firebase instead of using this.$fire

Removing those additional import statements solved the problem for me.

**Comment by lupas** [comment]
@tibs245 & @buwilliams are you guys sure you're not import Firebase somewhere else in your project?

This module imports Firebase for you, you shouldn't be importing Firebase yourself additionally.

**Comment by buwilliams** [comment]
@lupas As far as I can tell (manually going through all the code as well as grepping) the answer is no. All I need to do is `enablePersistence` to get this error. It may help to know this is not a SSR project.

```javascript
firestore: {
    enablePersistence: {
        synchronizeTabs: true
    },
    emulatorPort: process.env.NODE_ENV === 'development' ? 8080 : undefined,
}
```

**Comment by tibs245** [comment]
@lupas Yes I am sure. I have do search on all project file and I not found import.

**Comment by lupas** [comment]
Hey @tibs245 & @buwilliams 

Thanks for your inputs! Found the issue and fixed it with https://github.com/nuxt-community/firebase-module/commit/bc16b45a712e3c6400f7c9f8fa1af4233679b46c, will release in a new version asap.

Issue was that we called useEmulator before enablePersistence. Switching that order works perfectly now.

**Comment by lupas** [comment]
Fixed with [v7.5.0](https://github.com/nuxt-community/firebase-module/releases/tag/v7.5.0), thanks everyone!

**Comment by buwilliams** [comment]
@lupas I'm impressed you took the time to dig in and solve it. I'd say this is a `+1 for your character` sir. I hope you have an extra nice Friday!
